build.gradle

```diff
@@ -23,13 +23,17 @@ dependencies {
     implementation 'org.springframework.boot:spring-boot-starter-webflux'
     compileOnly 'org.projectlombok:lombok'
     annotationProcessor 'org.projectlombok:lombok'
+    testCompileOnly 'org.projectlombok:lombok'
+    testAnnotationProcessor 'org.projectlombok:lombok'
     testImplementation('org.springframework.boot:spring-boot-starter-test') {
         exclude group: 'org.junit.vintage', module: 'junit-vintage-engine'
     }
     testImplementation 'de.flapdoodle.embed:de.flapdoodle.embed.mongo'
-    testImplementation 'io.projectreactor:reactor-test'
```

Comment:
`fun getDeviceHealth(ipAddress: String, deviceResources: DeviceResources?,
                        device: Device):DeviceHealth {
        return if(deviceResources != null) {
            DeviceHealth(
                    ip = ipAddress,
                    ps = deviceResources.psuStatuses.map {
                        PsStatus(
                                name = "PSU${it.name.last()}",
                                status = it.getDescriptivePSStatus()
                        )
                    }.sortedBy { it.name },
                    fan = getFanStatus(device, deviceResources),
                    cpu = deviceResources.currentCpu,
                    ram = deviceResources.currentRam,
                    flash = deviceResources.currentFlash,
                    temperature = if (deviceResources.minTemperature != deviceResources.maxTemperature) {
                        listOf(TemperatureStatus(deviceResources.minTemperature, deviceResources.maxTemperature, deviceResources.currentTemperature))
                    } else listOf(),
                    totalFlashInBytes = deviceResources.totalFlashInBytes,
                    currentFlashInBytes = deviceResources.currentFlashInBytes,
                    totalRamInMB = deviceResources.totalRamInMB,
                    currentRamInMB = deviceResources.currentRamInMB,
                    temperaturePerModule = getTemperaturePerModule(device.modules, device.properties)
            )
        } else {
            DeviceHealth(ip = ipAddress)
        }
    }`